lib/http2/connection.c

```diff
@@ -1715,7 +1715,9 @@ static h2o_http2_conn_t *create_conn(h2o_context_t *ctx, h2o_hostconf_t **hosts,
         }},
     };
 
-    h2o_http2_conn_t *conn = (void *)h2o_create_connection(sizeof(*conn), ctx, hosts, connected_at, &callbacks);
+    void *_conn = h2o_mem_alloc_recycle(&h2o_mem_req_allocator);
```

Comment:
I'm not entirely sure if we need to recycle `h2o_http2_conn_t`. I would suspect that TLS handshake is so heavy that probably it's not worth recycling connection-level chunks that would be allocated alongside a TLS handshake.

If we are to reuse, I think we have to change the invocation of `free` as well?